cmd/etcd-mesos-scheduler/app.go

```diff
@@ -122,12 +124,16 @@ func main() {
 		}
 	}
 
+	if *advertiseAddress == "" {
+		*advertiseAddress = *address
```

Comment:
ack. i should have scrolled up! the `address` logic could easily be rolled into a `defaultAddress()` func since it doesn't depend on any internal vars. this `advertiseAddress` value is basically defaulting to the value of another flag. it's not quite duplicating existing logic because `PublishedAddress` below will always have a value now. I'd rather see that param unset if the `advertiseAddress` flag is unspecified. does that make sense?